engine.py

```diff
@@ -440,6 +446,29 @@ def __on_project_opened(self, opened_project, is_new):
         # change current engine context:
         sgtk.platform.change_context(ctx)
 
+    def __on_project_saved(self, saved_project):
+        workfiles_app = self.apps.get("tk-multi-workfiles2")
+        proj_mgr_app = self.apps.get("tk-mari-projectmanager")
+        if not workfiles_app or not proj_mgr_app:
+            self.log_error("Unable to find workfiles or projectmanager app. Not exporting msf file.")
+            return
+
+        fields = self.context.as_template_fields()
+
+        # use project name to obtain "name" field
+        project_name_template = proj_mgr_app.get_template("template_new_project_name")
+        fields.update(project_name_template.get_fields(saved_project.name()))
+
+        # use project metadata to get the "version" field
+        # assume 1 if no version is set yet
```

Comment:
For backwards compatibility with already created projects that have no version set.

Reply:
I'd think you'd want to hijack the File Save dialog to open tk-multi-workfiles instead, in which case `version` will always be set. Not that I'm opposed to this, but I would check out how other engines handle this situation and mirror that functionality.